### PR TITLE
[RLLib] Record framework and algorithm used by an RLlib run.

### PR DESCRIPTION
**Pull request by gjoliver**
## Why are these changes needed?

Automatically record framework and algorithm used by RLlib jobs.
For better planning.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [*] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by gjoliver** [comment]
@JiahaoYao how do I know if this is working??

**Review by ericl — APPROVED** [review]

**Comment by gjoliver** [comment]
@jjyao, sorry tagged the wrong person, any idea how to test this a bit locally?

File: python/ray/_private/usage/usage_lib.py [hunk 1]
```diff
@@ -244,6 +244,8 @@ def _put_library_usage(library_usage: str):
 class TagKey(Enum):
     _TEST1 = auto()
     _TEST2 = auto()
+    _RLLIB_FRAMEWORK = auto()
```
Comment (jjyao) [inline python/ray/_private/usage/usage_lib.py:247]: You don't have to start with `_` :)  `_TEST1` is meant for private testing so it starts with `_`.

File: rllib/algorithms/algorithm.py [hunk 2]
```diff
@@ -2562,6 +2565,15 @@ def _compile_iteration_results(
...
         return type(self).__name__
 
+    def _record_usage(self, config):
+        """Record the framework and algorithm used.
+
+        Args:
+            config: Algorithm config dict.
+        """
+        record_extra_usage_tag(TagKey._RLLIB_FRAMEWORK, config["framework"])
+        record_extra_usage_tag(TagKey._RLLIB_ALGORITHM, self.__class__.__name__)
```
Comment (jjyao) [inline rllib/algorithms/algorithm.py:2575]: Is it possible that multiple frameworks or algorithms are used in one cluster?

File: python/ray/_private/usage/usage_lib.py (hunk 1 above)
Comment (gjoliver) [reply python/ray/_private/usage/usage_lib.py:247]: oh, copy & paste is the only way I know :)
changed.

File: rllib/algorithms/algorithm.py (hunk 2 above)
Comment (gjoliver) [reply rllib/algorithms/algorithm.py:2575]: there should only be 1 algorithm in training.
thanks for the good question.

**Comment by gjoliver** [comment]
I also tested this with your instructions, and it works really well!!
thanks a lot for the nice feature.

**Review by thomasdesr — APPROVED** [review]
Boop for the `USER_DEFINED` change

**Comment by kouroshHakha** [comment]
What is the expected behavior if a user submits a job that creates different trainers for some weird use-case? e.g. Multi-agent where one agent is trained with DQN and another one with PPO?

**Review by kouroshHakha — APPROVED** [review]

**Comment by gjoliver** [comment]
> ates different trainers for some weird use-case? e.g. Multi-agent where one agent is trained with DQN and another one with PPO?

very good question. doesn't seem like it's supported today.
let's chat with the team to figure out a right fix.

**Comment by kouroshHakha** [comment]
Stamped for now, here is the conclusion as discussed offline: This telemetry is a per-cluster key that will get overwritten if you have a new algorithm. For those rare use-cases the returned behavior would not be exactly correct.

**Pull request merged**